File: JavaScript/Math-Array-String/64.js
// 2965. Find Missing and Repeated Values

/**
 * @param {number[][]} grid
 * @return {number[]}
 */
var findMissingAndRepeatedValues = function(grid) {
    const n = grid.length; // Grid size (n x n)
    const totalNumbers = n * n;
    
    // Flatten the 2D grid into a 1D array
    const flatArray = grid.flat();
    
    // Create an array to count occurrences of each number
    const count = Array(totalNumbers + 1).fill(0);
    for (const num of flatArray) {
        count[num]++;
    }
    
    let repeated = -1, missing = -1;
    // Check for repeated and missing numbers
    for (let i = 1; i <= totalNumbers; i++) {
        if (count[i] === 2) {
            repeated = i;
        } else if (count[i] === 0) {
            missing = i;
        }
    }
    
    return [repeated, missing];
};

// Example Usage:
const grid1 = [[1, 3], [2, 2]];
console.log(findMissingAndRepeatedValues(grid1)); // Output: [2, 4]

const grid2 = [[9, 1, 7], [8, 9, 2], [3, 4, 6]];
console.log(findMissingAndRepeatedValues(grid2)); // Output: [9, 5]

File: Python/Math-Array-String/64.py
# 2965. Find Missing and Repeated Values

from typing import List

class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        total_numbers = n * n
        flat_list = [num for row in grid for num in row]
        
        # Create a count array to track the frequency of each number
        count = [0] * (total_numbers + 1)
        for num in flat_list:
            count[num] += 1
        
        # Find the repeated (a) and missing (b) numbers
        a = b = -1
        for i in range(1, total_numbers + 1):
            if count[i] == 2:
                a = i
            elif count[i] == 0:
                b = i
        
        return [a, b]
    

sol = Solution()
output = sol.findMissingAndRepeatedValues([[9,1,7],[8,9,2],[3,4,6]])
print(output) # # Output: [9, 5]